src/main/java/com/epam/aidial/core/util/ProxyUtil.java

```
@@ -83,7 +87,7 @@ private static int contentLength(MultiMap header, int defaultValue) {
         return defaultValue;
     }
 
-    public static void collectAttachedFiles(ObjectNode tree, ApiKeyData apiKeyData) {
+    public static void collectAttachedFiles(ObjectNode tree, ApiKeyData apiKeyData, Function<String, Boolean> checkAccessFn) throws Exception {
```

Consumer<String> accessChecker?

---

Do you want to throw an exception in the consumer auction directly?

---

Yes

---

agreed to keep it as is